### PR TITLE
composite_filter: Improve performance by using absl::visit with lambda overloading 

### PR DESCRIPTION
**Pull request by tyxia**
std::visit with lambda overload 
 - significantly improve the performance of std::variant by eliminating the if-else/switch runtime checks and get function (confirmed with benchmark)
 - more type-safe since compiler will  make sure it is alternative exhaustive 
 - make code concise

Signed-off-by: tyxia [tyxia@google.com](mailto:tyxia@google.com)

### CODE REVIEW

**Comment by tyxia** [comment]
gently ping @snowp :)
This PR should be straightforward

File: source/extensions/filters/http/composite/filter.cc [hunk 1]
```diff
@@ -20,6 +20,12 @@ RValT delegateFilterActionOr(FilterPtrT& filter, FuncT func, RValT rval, Args&&.
 
   return rval;
 }
+
+// Own version of lambda overloading since std::overloaded is not available to use yet.
+template <class... Ts> struct Overloaded : Ts... { using Ts::operator()...; };
+
+template <class... Ts> Overloaded(Ts...) -> Overloaded<Ts...>;
```
Comment (snowp) [inline source/extensions/filters/http/composite/filter.cc:27]: I think this has been defined in several places in the repo with different names, might be nice to have this in a shared location but not necessary for this PR, I'll file an issue

**Review by snowp — APPROVED** [review]
Thanks!

File: source/extensions/filters/http/composite/filter.cc (hunk 1 above)
Comment (tyxia) [reply source/extensions/filters/http/composite/filter.cc:27]: Sure, I have filed #26442 and assigned to myself.

Comment (snowp) [reply source/extensions/filters/http/composite/filter.cc:27]: Thanks!

**Pull request merged**